libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -0,0 +1,65 @@
+import { getConfig, localizeLink } from '../../../utils/utils.js';
+
+export function toFragment(htmlStrings, ...values) {
+  const templateStr = htmlStrings.reduce((acc, htmlString, index) => {
+    if (values[index] instanceof HTMLElement) {
+      return `${acc + htmlString}<elem ref="${index}"></elem>`;
+    }
+    return acc + htmlString + (values[index] || '');
+  }, '');
+
+  const fragment = document.createRange().createContextualFragment(templateStr).children[0];
+
+  Array.prototype.map.call(fragment.querySelectorAll('elem'), (replaceable) => {
+    const ref = replaceable.getAttribute('ref');
+    replaceable.replaceWith(values[ref]);
+  });
+
+  return fragment;
+}
+
+// TODO this is just prototyped
+export const getFedsPlaceholderConfig = () => {
+  const { locale, miloLibs, env } = getConfig();
+  let libOrigin = 'https://adobe.com/libs';
+  if (window.location.origin.includes('localhost')) {
+    libOrigin = `${window.location.origin}/libs`;
+  }
+
+  if (window.location.origin.includes('.hlx.')) {
+    const baseMiloUrl = env.name === 'prod'
+      ? 'https://main--milo--adobecom.hlx.live'
+      : 'https://main--milo--adobecom.hlx.page';
+    libOrigin = miloLibs || `${baseMiloUrl}/libs`;
+  }
+
+  return {
+    locale: {
+      ...locale,
+      contentRoot: `${libOrigin}/feds${locale.prefix}`,
+    },
+  };
+};
+
+export function getAnalyticsValue(str, index) {
+  if (typeof str !== 'string' || !str.length) return str;
+
+  let analyticsValue = str.trim().replace(/[^\w]+/g, '_').replace(/^_+|_+$/g, '');
+  analyticsValue = typeof index === 'number' ? `${analyticsValue}-${index}` : analyticsValue;
+
+  return analyticsValue;
+}
+
+export function decorateCta({ elem, type = 'primaryCta', index } = {}) {
+  const modifier = type === 'secondaryCta' ? 'secondary' : 'primary';
+
+  return toFragment`
+    <div class="feds-cta-wrapper">
+      <a 
+        href="${localizeLink(elem.href)}"
```

Comment:
do you really have to call localizeLink? 
i thought that all links are localized in the utils.js file

Reply:
The content for the global-navigation comes through another request, thus goes through another pipeline.

Assuming my page URL is https://www.adobe.com/testPath and I add a link with this `href` on the page, then keeping the call to `localizeLink` means the final `href` value will be `.testPath`. Removing the call to `localizeLink` would mean that the final `href` value will be `https://www.adobe.com/testPath`. What would be the desired effect between the two?

Keeping the call to `localizeLink` means that

Reply:
why not have the content go through the normal pipeline though? That way there's much less chance of forgetting to decorate stuff properly ...japanese word wrap, link localization, etc. etc. 
You'd call the normal pipeline by calling loadArea on your loaded html